app/services/utils/collection.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional, Dict, Any
+
+import requests
+
+
+def find_collection_url(base_url: str, collection_id: str) -> Optional[str]:
+    response = requests.get(base_url)
+    response.raise_for_status()
+    root_data = response.json()
+
+    collection_url = None
+    for link in root_data.get("links", []):
+        href = link.get("href")
+        if (
+            isinstance(href, str)
+            and link.get("rel") == "child"
+            and collection_id in href
+        ):
+            collection_url = href
+            break
+
+    if not collection_url:
+        raise ValueError(
+            f"No collection found for id at URL: {base_url}/collections/{collection_id}"
+        )
+
```

Comment:
why did you just dismiss the previous comment? I don't see a reason to keep looking in the links object instead of just querying the direct collection url

Reply:
the same for the items